src/dotnet/Core/DependencyInjection/ServiceCollectionExt.cs

```diff
@@ -26,4 +26,74 @@ public static class ServiceCollectionExt
         this IServiceCollection services,
         Action<Dictionary<Type, Type>> typeMapBuilder)
         => services.AddSingleton(_ => new TypeMap<TScope>(typeMapBuilder));
+
+    public static ServiceDescriptor ChangeLifetime(
+        this ServiceDescriptor serviceDescriptor,
+        ServiceLifetime serviceLifetime)
+    {
+        if (serviceDescriptor.ImplementationFactory != null)
+            return new ServiceDescriptor(serviceDescriptor.ServiceType,
+                serviceDescriptor.ImplementationFactory,
+                serviceLifetime);
```

Comment:
please reformat with Rider and add new lines

Reply:
done